### ISSUE TITLE
[VETS-API] Gem Update: Bump savon to latest (2.15.x)

### ISSUE THREAD

**Issue by ericboehs** [department-of-veterans-affairs/va.gov-team#74433]
## Issue Description
Savon is several minor versions behind now (2.12, released 2020, while current is 2.15). We've attempted to update in the past, but ran into issues, as described in the blocking ticket. Bumping savon requires re-recording a bunch of cassettes, which is complicated (and nobody seems to have the knowledge to correctly execute). If we can re-record cassettes, we can bump the gem easily. To do this, we must first bump savon in `bgs_ext` and release a new version (separate ticket)

NOTE: According to Tom Harrison, the 3 specs from BGS should actually be REMOVED in the next month or so, we may not need to worry about these at all (and I think these would have been the hardest part). Woohoo!

Cassettes requiring re-recording:
* ~~52~~ 36 total examples failing across 9 specs
* ~~28~~ ~22 total cassettes need recording, some being used my multiple specs/examples
* Essentially 3 teams, with 2-3 methods of re-recording (SSH, RI's) are responsible for the following:

- [x] ~~`./spec/controllers/v0/profile/ch33_bank_accounts_controller_spec.rb` (2 specs)~~
  - [x] Cassettes (note: each example uses multiple specs... re-recording will be extra difficult):
    - [x] `bgs/service/find_ch33_dd_eft`
    - [x] `bgs/service/update_ch33_dd_eft`
    - [x] `bgs/ddeft/find_bank_name_valid`
  * Team: [vfs-authenticated-experience-backend](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-authenticated-experience-backend) (for reaching out to bgs, per Lihan)
  * Method: [SSH to BGS](https://vfs.atlassian.net/wiki/spaces/pilot/pages/2893611114/VCR+debugging#Re-recording-VCR-Cassettes)
- [x] `./spec/lib/bgs/service_spec.rb` ~~(13 specs)~~ (1 spec)
  - [x] Cassettes (note: each example uses multiple specs... re-recording will be extra difficult):
    - [x] `bgs/service/find_rating_data`
    - [x] ~~`bgs/service/find_ch33_dd_eft`~~
    - [x] ~~`bgs/service/find_ch33_dd_eft_no_icn`~~
    - [x] ~~`bgs/service/find_ch33_dd_eft_no_bank_info`~~
    - [x] ~~`bgs/service/update_ch33_dd_eft`~~
    - [x] ~~`bgs/ddeft/find_bank_name_valid`~~
    - [x] ~~`bgs/ddeft/find_bank_name_invalid_routing`~~
  * Team: [vfs-authenticated-experience-backend](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-authenticated-experience-backend) (for reaching out to bgs, per Lihan), but maybe [benefits-dependents-management](https://github.com/orgs/department-of-veterans-affairs/teams/benefits-dependents-management) (codeowners)
  * Method: [SSH to BGS](https://vfs.atlassian.net/wiki/spaces/pilot/pages/2893611114/VCR+debugging#Re-recording-VCR-Cassettes)
- [x] ~~`./spec/models/bank_name_spec.rb` (1 spec)~~
  - [x] Cassettes:
    - [x] `bgs/ddeft/find_bank_name_valid` (same as in first few items)
  * Team: [vfs-authenticated-experience-backend](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-authenticated-experience-backend)
  * Method: [SSH to BGS](https://vfs.atlassian.net/wiki/spaces/pilot/pages/2893611114/VCR+debugging#Re-recording-VCR-Cassettes)
- [x] `./spec/lib/form1010_ezr/service_spec.rb` (8 specs)
  - [x] Cassettes:
    - [x] `form1010_ezr/authorized_submit` (Note: this one broke everything - might require an update to wsdl file or something)
    - [x] `form1010_ezr/authorized_submit_with_mexican_province` (probably same issue as above)
    - [x] `form1010_ezr/authorized_submit_with_next_of_kin_and_emergency_contact`
    - [x] `form1010_ezr/authorized_submit_with_tera`
    - [x] `form1010_ezr/authorized_submit_with_attachments`
    - [x] `form1010_ezr/authorized_submit_with_non_pdf_attachment`
  * Team: [vfs-10-10](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-10-10)
  * Method: [Record in RI's](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/tutorials/record_vcr.md)
- [x] `./spec/lib/hca/service_spec.rb` (11 specs)
  - [x] Cassettes:
    - [x] `hca/short_form`
    - [x] `hca/demographic_no`
    - [x] `hca/medicare_claim_num`
    - [x] `hca/sigi`
    - [x] `hca/tera`
    - [x] `hca/submit_with_attachment`
    - [x] `hca/submit_with_attachment_jpg`
    - [x] `hca/health_check`
    - [x] `hca/health_check_downtime`
  * Team: [vfs-10-10](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-10-10)
  * Method: [Record in RI's](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/tutorials/record_vcr.md)
- [x] `./spec/requests/form1010_ezrs_request_spec.rb` (4 specs)
  - [x] Cassettes:
    - [x] `form1010_ezr/authorized_submit` (breaks everything, duplicate)
    - [x] `form1010_ezr/authorized_submit_with_mexican_province`
    - [x] `form1010_ezr/authorized_submit_with_next_of_kin_and_emergency_contact`
    - [x] `form1010_ezr/authorized_submit_with_tera`
    - [x] `form1010_ezr/authorized_submit_async` - now just this one?
  * Team: [vfs-10-10](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-10-10) (Joshua Drumm)
  * Method: [Record in RI's](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/tutorials/record_vcr.md)
- [x] `./spec/requests/health_care_applications_request_spec.rb` (4 specs)
  - [x] Cassettes:
    - [x] `bgs/service/find_rating_data`
    - [x] `hca/health_check`
    - [x] `hca/submit_anon`
    - [x] `hca/submit_auth`
  * Team: [vfs-10-10](https://github.com/orgs/department-of-veterans-affairs/teams/vfs-10-10)
  * Method: [Record in RI's](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/tutorials/record_vcr.md)
- [x] ~~`./spec/lib/preneeds/service_spec.rb` (2 specs)~~ - Kenny Santiago (BAH) is "getting ready to submit a PR" to remove specs (8/22)
  - [x] Cassettes:
    - [x] `preneeds/burial_forms/burial_form_with_attachments`
    - [x] `preneeds/burial_forms/creates_a_pre_need_burial_form`
  * Team: [mbs-core-team](https://github.com/orgs/department-of-veterans-affairs/teams/mbs-core-team) (?)
  * Method: Unclear. Nobody seems to have ownership. You can connect to the upstream service via SSH/staging fwdproxy/port 4439. This will likely require some more hacking, so point accordingly.
- [x] `./spec/requests/swagger_spec.rb` (7 specs)
  - [x] Cassettes:
    - [x] `bgs/service/find_rating_data`
    - [x] `/v0/health_care_applications/healthcheck` - maybe the same as `hca/healthcheck`?
    - [x] `hca/submit_anon`
    - [x] `form1010_ezr/authorized_submit_with_es_dev_uri`
    - [x] `bgs/service/find_ch33_dd_eft`/ `bgs/service/update_ch33_dd_eft` / `bgs/ddeft/find_bank_name_valid`
  * Team: various, though I think all are the same as above
  * Method: varies, but same as others

---
## Tasks
- [x] Bump `bgs_ext` and `savon` gems
- [x] Merge in [PR from Tom Harrison](https://github.com/department-of-veterans-affairs/vets-api/pull/17785)
- [x] Merge in HCA and Form1010 PRs, [#74461](https://github.com/department-of-veterans-affairs/va.gov-team/issues/74461)

## Acceptance Criteria
- [ ] Deployed to Prod, without breaking everything this time.

**Comment by ericboehs** [comment]
https://github.com/department-of-veterans-affairs/va.gov-team/issues/87359

**Comment by tpharrison** [comment]
@ryan-mcneil

I've reviewed the VCR cassettes related to CH33, and they can be removed.  Here is some background that explains why we can do this...

Until recently. the Profile page on VA.gov allowed a user to manage their Comp and Pen direct deposit and Education Benefits  direct deposit information separately.  In other words, a user could have two different direct deposit accounts (one for C&P and one for EDU Benefits).

In February 2024, the Authenticated Experience team was notified that these two accounts would be merged into a single account going forward.  In addition, BGS would no longer be called directly from vets-api to manage Education Benefits direct deposit information; instead the Lighthouse Direct Deposit API would be used to manage a single direct deposit account that would be used for both C&P and Education Benefits.  As a result, the `Profile::CH33BankAccountsController`, which handled Education Benefits, will no longer be used.

On May 31st, [this PR](https://github.com/department-of-veterans-affairs/vets-api/pull/16893) was merged to disable the `Profile::CH33BankAccountsController`.  It uses a feature flag named `profile_show_direct_deposit_single_form_edu_downtime` , that when enabled, will return a `403 code with the following message - `This endpoint is deprecated and will be removed soon`.

[These logs](https://vagov.ddog-gov.com/logs?query=env%3Aeks-prod%20%40http.url_details.path%3A%2Fv0%2Fprofile%2Fch33_bank_accounts%2A%20&agg_m=count&agg_m_source=base&agg_q=%40http.status_code&agg_q_source=base&agg_t=count&cols=host%2Cservice%2C%40http.status_code%2C%40payload.user_uuid&fromUser=true&messageDisplay=inline&refresh_mode=sliding&sort=time&spanID=336994431213512540&storage=hot&stream_sort=%40payload.user_uuid%2Cdesc&top_n=10&top_o=top&viz=sunburst&x_missing=true&from_ts=1719154523464&to_ts=1721746523464&live=true) show that although the controller is being called in production, it has essentially been disabled. 

Here is a high-level overview of the CH33 related dependencies:

```mermaid
classDiagram
    Service <|-- ch33_bank_accounts_controller
    Service <|-- ch33_bank_accounts_controller_spec
    Service <|-- service_spec
    Service <|-- swagger_spec
    Service: +get_ch33_dd_eft_info()
    Service: +find_ch33_dd_eft()
    Service: +update_ch33_dd_eft()
    class ch33_bank_accounts_controller{
      +index()
      +update()
    }
    class ch33_bank_accounts_controller_spec{
      # multiple specs
    }
    class service_spec{
      # multiple specs
    }
    class swagger_spec{
      # multiple specs
    }
```

Because of the recent migration to Lighthouse, the four dependencies above (and any corresponding cassettes) can be removed.

I will remove them in the `rm-bump-bgs_ext-and-savon` branch and reference the PRs.

**Comment by ryan-mcneil** [comment]
PRs up:
[Removing ch33 files](https://github.com/department-of-veterans-affairs/vets-api/pull/18054)
[Main PR](https://github.com/department-of-veterans-affairs/vets-api/pull/18051) for the savon/httpi/connect_vbms/bgs_ext bump
[PR for HCA cassettes](https://github.com/department-of-veterans-affairs/vets-api/pull/18115) (to be merged into Main PR)

**Comment by ryan-mcneil** [comment]
possible culprit to investigate for savon change: https://github.com/savonrb/savon/pull/943/files

**Comment by ryan-mcneil** [comment]
I repurposed this ticket a bit. As we had [another ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/74461) "investigating the re-recording of cassettes", I allowed that to encompass the majority of the work that was done this week. This ticket now will now include the merging of the various other tasks (merging in PRs from other services, above, into the [Parent PR](https://github.com/department-of-veterans-affairs/vets-api/pull/18051), which is actually bumping the gems as described in this ticket.

Over the next week I will work with those teams to merge this PR, monitoring in both staging and production as not have any repeats of last time
